tools/boards.txt.py

```diff
@@ -1406,44 +1420,93 @@ def all_flash_map ():
     f8m  = collections.OrderedDict([])
     f16m = collections.OrderedDict([])
 
-    #                      flash(KB) spiffs(KB)
+    global c_flash_map
+    c_flash_map = collections.OrderedDict([])
 
-    f1m.update( flash_map(    1024,      64 ))
+    #                      flash(KB) spiffs(KB) confname(C)
+
+    f1m.update( flash_map(    1024,      64, 'OTA_FS' ))
     f1m.update( flash_map(    1024,     128 ))
     f1m.update( flash_map(    1024,     144 ))
     f1m.update( flash_map(    1024,     160 ))
     f1m.update( flash_map(    1024,     192 ))
     f1m.update( flash_map(    1024,     256 ))
-    f1m.update( flash_map(    1024,     512 ))
-    f1m.update( flash_map(    1024))
+    f1m.update( flash_map(    1024,     512, 'MAX_FS' ))
+    f1m.update( flash_map(    1024,       0, 'NO_FS'  ))
 
-    f2m.update( flash_map(  2*1024,      64 ))
+    f2m.update( flash_map(  2*1024,      64, 'OTA_FS' ))
```

Comment:
Personally, I would like to see a bit more FS with the OTA_FS for 2M flash.
256k still leaves roughly 850k of sketch + OTA.
But 64k makes almost no sense for a file system as you still need to have some blocks free (at least with SPIFFS) to make it work.

Reply:
256K for 2MB flash chips it is. It is a fair choice (leaving 1.75MB for sketch + gzipped OTA).
Please head to OP, which has now links to updated documentation.